src/components/navbar/Navbar.vue

```diff
@@ -0,0 +1,99 @@
+<template>
+    <nav id="zek-navbar" :class="customClass" :style="styleObj">
+        <ul class="zek-navbar-menu" v-if="tabs && tabs.length">
+            <li class="zek-navbar-menu-item" v-for="(tab, index) in tabs" :key="index">
+                <ZekButton
+                    :label="tab.label"
+                    :url="tab.url"
+                    theme= "bootstrap"
+                    customClass="nav-tab"
+                    :styleObj="tabStyle"
+                    :disabled="tab.disabled"
+                    :icon="tab.icon"
+                    :active="tab.active"
+                    @onClick="handleTabClick(index)"
+                ></ZekButton>
+
+                <ZekNavbar
+                    v-if="tab.tabs && tab.showNested"
+                    :tabs="tab.tabs"
+                    :styleObj="nestedStyleObj"
+                    :tabStyle="nestedTabStyle"
+                    class="position-absolute mt-3"
+                />
+            </li>
+        </ul>
+    </nav>
+</template>
+<script>
+import ZekButton from "../action-button/ActionButton.vue";
+
+export default {
+    name: "ZekNavbar",
+    components: {
+        ZekButton
+    },
+    props: {
+        tabs: {
+            type: Array,
+            required: true
+        },
+        styleObj: {
+            type: Object,
+            default: () => ({})
+        },
+        customClass: {
+            type: String,
+            default: "zek-navbar-wrapper"
+        },
+        tabStyle: {
+            type: Object,
+            default: () => ({})
+        },
+        nestedTabStyle: {
+            type: Object,
+            default: () => ({})
+        },
+        nestedStyleObj: {
+            type: Object,
+            default: () => ({})
+        },
+    },
+    data() {
+        return {};
+    },
+    methods: {
+        handleTabClick(index) {
+            this.tabs.forEach((tab, i) => {
+                this.$set(tab, "showNested", i === index );
+                this.$set(tab, "active", i === index );
+            });
```

Comment:
1. Why are we using `$set` here, can the variables not be referred directly? or would that affect reactivity?
2. What happens if another tab was already selected does it hide it? or not?
3. An event should be emitted here that should also be reflected in the nested navbars with `v-on`

Reply:
1.I used $set to maintain the reactivity since I am trying to track a prop in nested array
2.yes if another was already selected it hide it only the selected tab will be active
3.alright